pyHS100/pyHS100.py

```diff
@@ -57,6 +63,39 @@ def __init__(self, ip_address, protocol=None):
             protocol = TPLinkSmartHomeProtocol()
         self.protocol = protocol
 
+        self.schedule = schedule.Schedule(self, self._query_helper)
+        self.countdown = countdown.Countdown(self, self._query_helper)
+        self.antitheft = antitheft.AntiTheft(self, self._query_helper)
+
+    def new_schedule(self):
+        """
+        Used by Schedule instance when it gets deleted.
+
+        :return: None
+        :rtype: None
+        """
+
+        self.schedule = schedule.Schedule(self, self._query_helper)
+
+    def new_countdown(self):
+        """
+        Used by Countdown instance when it gets deleted.
+
+        :return: None
+        :rtype: None
+        """
+
+        self.countdown = countdown.Countdown(self, self._query_helper)
+
+    def new_antitheft(self):
+        """
+        Used by AntiTheft instance when it gets deleted.
+        :return: None
+        :rtype: None
+        """
+
+        self.antitheft = antitheft.AntiTheft(self, self._query_helper)
+
```

Comment:
the reason for this being done is because if you wanted to delete all scheduled/countdown/antitheft rules you would simply delete the instance from the device.

    del device_instance.antitheft

would cause the following code to run removing all antitheft rules form the device. but the antitheft instance still needs to exist so if the user wanted to make new rules

        def __del__(self):
            """
            Deletes all rules.

            :return: None
            :rtype: None
            :raises SmartPlugException: on error
            """

            self._query_helper(
                "anti_theft",
                "delete_all_rules"
            )
            self._device.new_antitheft()